libs/frontend/feature-build/src/lib/code-editor-plugins.ts

```diff
@@ -0,0 +1,56 @@
+import { schemaCode } from '@pubstudio/frontend/util-edit-text'
+import { Keys } from '@pubstudio/frontend/util-key-listener'
+import hljs from 'highlight.js/lib/core'
+import { keymap } from 'prosemirror-keymap'
+import { Fragment, Slice } from 'prosemirror-model'
+import { Plugin } from 'prosemirror-state'
+import { highlightPlugin } from './prosemirror-highlighthjs-plugin'
+
+// Manually create a new slice instead of using the third parameter `slice` from
+// `handlePaste` because new line characters will be ignored for unknown reason
+// when using it to do content replacement in the editor state.
+const transformPasteContentPlugin = new Plugin({
```

Comment:
Hi @sampullman 

I think it's finally solved! For some reason, new line characters from an outer source (not copied within a prosemirror editor) will be ignored when pasted to to editor. I'm not sure if it's something related to how we use the prosemirror editor or how we define the schema 🤔?

Manually create a new slice and inserting it back to the editor seems to work here.